src/devices/Ads1115/Ads1115.cs

```diff
@@ -47,10 +47,10 @@ public class Ads1115 : IDisposable
         /// <param name="measuringRange">Programmable Gain Amplifier</param>
         /// <param name="dataRate">Data Rate</param>
         /// <param name="deviceMode">Initial device mode</param>
-        public Ads1115(I2cDevice i2cDevice, InputMultiplexer inputMultiplexer = InputMultiplexer.AIN0, MeasuringRange measuringRange = MeasuringRange.FS4096, DataRate dataRate = DataRate.SPS128,
-            DeviceMode deviceMode = DeviceMode.Continuous)
+        public Ads1115(I2cDevice i2cDevice, InputMultiplexer inputMultiplexer = InputMultiplexer.AIN0, MeasuringRange measuringRange = MeasuringRange.FS4096,
+            DataRate dataRate = DataRate.SPS128, DeviceMode deviceMode = DeviceMode.Continuous)
         {
-            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException($"{nameof(i2cDevice)} cannot be null");
```

Comment:
You're only supposed to provide the name of the argument.
See https://docs.microsoft.com/en-us/dotnet/api/system.argumentnullexception.-ctor?view=netcore-3.1#System_ArgumentNullException__ctor_System_String_
Otherwise if you want to add a message use the second parameter, but the above message would pretty much already be generated (and probably localized too). See
https://docs.microsoft.com/en-us/dotnet/api/system.argumentnullexception.-ctor?view=netcore-3.1#System_ArgumentNullException__ctor_System_String_System_String_

Reply:
Got it. I noticed that there are two patterns that are pervasive in the codebase. I apparently picked the wrong one. I didn't Have this background. I'll fix them all on a second PR.

Reply:
I added another commit with many instances of `ArgumentException` fixed, well beyond the initial scope of the PR. Good call.